### ISSUE TITLE
kernel: -N filter error when replaying trace data

### ISSUE THREAD

**Issue by Taeung** [namhyung/uftrace#184]
I found a error regarding a `-N` option with a kernel function.
As you can see below, even if `smp_irq_work_interrupt` kernel symbol exists,
-N option don't work when replaying trace data.

Is it a bug ?

If using -N option with a user function,
there aren't any problems.

```
$ uftrace replay -N smp_irq_work_interrupt@kernel -d uftrace.data.nmap_kern.old/
fstack: /home/taeung/git/uftrace/utils/fstack.c:368:fstack_setup_filters
 ERROR: failed to set filter or trigger: !smp_irq_work_interrupt@kernel

$ uftrace replay -d uftrace.data.nmap_kern.old/
# DURATION    TID     FUNCTION
            [10756] | _GLOBAL__sub_I_MacTable() {
            [10756] |   __do_page_fault() {
            [10756] |     smp_irq_work_interrupt() {

```

**Comment by namhyung** [comment]
Hmm.. ok.  I can reproduce it and I pushed the fix to [check/filter-fix](https://github.com/namhyung/uftrace/compare/check/filter-fix).  Could you please confirm it?

**Comment by Taeung** [comment]
Yep, it works without problems.
I tested the branch you said.
Thanks for your quick work @namhyung 

```
### Before kernel filtering has some problems.

$ uftrace record -K 5 ./simple_hello

$ uftrace replay -N syscall_slow_exit_work@kernel 
fstack: /home/taeung/git/uftrace/utils/fstack.c:368:fstack_setup_filters
 ERROR: failed to set filter or trigger: !syscall_slow_exit_work@kernel
```
```
### After rebuilding uftrace on the check/filter-fix branch, it ordinarily works.

$ uftrace replay -N syscall_slow_exit_work@kernel  -N smp_irq_work_interrupt@kernel -N syscall_trace_enter_phase1@kernel
# DURATION    TID     FUNCTION
   1.114 us [16803] | __monstartup();
   1.046 us [16803] | __cxa_atexit();
            [16803] | main() {
            [16803] |   printf() {
            [16803] |     sys_newfstat() {
            [16803] |       vfs_fstat() {
            [16803] |         __fdget_raw() {
            [16803] |           __fget_light() {
   4.035 us [16803] |           } /* __fget_light */
   4.682 us [16803] |         } /* __fdget_raw */
            [16803] |         vfs_getattr() {
            [16803] |           security_inode_getattr() {
   0.797 us [16803] |             apparmor_inode_getattr();
   1.970 us [16803] |           } /* security_inode_getattr */
            [16803] |           vfs_getattr_nosec() {
   0.310 us [16803] |             generic_fillattr();
   1.091 us [16803] |           } /* vfs_getattr_nosec */
   4.061 us [16803] |         } /* vfs_getattr */
   9.656 us [16803] |       } /* vfs_fstat */

```

**Comment by namhyung** [comment]
Merged: d116982
